projects/plugins/jetpack/class.jetpack.php

```diff
@@ -2666,34 +2666,23 @@ public static function get_module_path( $slug ) {
 	 * plugins on the WordPress plugins page.
 	 */
 	public static function get_module( $module ) {
-		$headers = array(
-			'name'                      => 'Module Name',
-			'description'               => 'Module Description',
-			'sort'                      => 'Sort Order',
-			'recommendation_order'      => 'Recommendation Order',
-			'introduced'                => 'First Introduced',
-			'changed'                   => 'Major Changes In',
-			'deactivate'                => 'Deactivate',
-			'free'                      => 'Free',
-			'requires_connection'       => 'Requires Connection',
-			'requires_user_connection'  => 'Requires User Connection',
-			'auto_activate'             => 'Auto Activate',
-			'module_tags'               => 'Module Tags',
-			'feature'                   => 'Feature',
-			'additional_search_queries' => 'Additional Search Queries',
-			'plan_classes'              => 'Plans',
-		);
-
 		static $modules_details;
+
 		$file = self::get_module_path( self::get_module_slug( $module ) );
 
 		if ( isset( $modules_details[ $module ] ) ) {
 			$mod = $modules_details[ $module ];
 		} else {
+			$mod = jetpack_get_module_info( $module );
 
-			$mod = self::get_file_data( $file, $headers );
 			if ( empty( $mod['name'] ) ) {
-				return false;
+				// Try to get the module info from the file as a fallback.
+				$mod = self::get_file_data( $file, jetpack_get_all_module_header_names() );
```

Comment:
I included this fallback because the `Jetpack::get_available_modules` method used the module file name in the call to `Jetpack::get_module`. I've submitted PR #20185 to change that call to use the module slug. I haven't been able to find any other cases where the module file name is used (rather than the module slug), but it's possible that they exist.